cluster/main.go

```diff
@@ -0,0 +1,192 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	ds "github.com/ipfs/go-datastore"
+	logging "github.com/ipfs/go-log"
+	ipfscluster "github.com/ipfs/ipfs-cluster"
+	"github.com/ipfs/ipfs-cluster/allocator/ascendalloc"
+	"github.com/ipfs/ipfs-cluster/allocator/descendalloc"
+	"github.com/ipfs/ipfs-cluster/config"
+	"github.com/ipfs/ipfs-cluster/consensus/crdt"
+	"github.com/ipfs/ipfs-cluster/informer/disk"
+	"github.com/ipfs/ipfs-cluster/informer/numpin"
+	"github.com/ipfs/ipfs-cluster/monitor/pubsubmon"
+	"github.com/ipfs/ipfs-cluster/observations"
+	"github.com/ipfs/ipfs-cluster/pintracker/maptracker"
+	"github.com/ipfs/ipfs-cluster/pintracker/stateless"
+	host "github.com/libp2p/go-libp2p-host"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+var log = logging.Logger("tex-cluster")
+
+func InitCluster(repoPath, listenAddr string) error {
+	cfgMgr, cfgs := makeClusterConfigs()
+	err := cfgMgr.Default()
+	if err != nil {
+		return err
+	}
+
+	if listenAddr != "" {
+		addr, err := ma.NewMultiaddr(listenAddr)
+		if err != nil {
+			return err
+		}
+		cfgs.ClusterCfg.ListenAddr = addr
+	}
+
+	return cfgMgr.SaveJSON(ConfigPath(repoPath))
```

Comment:
I think you don't need to save (it's writing or overwriting to disk). Might be useful to debug what actual config you are generating.

Reply:
I'll have to double check, but I think this is the first time it's written